source/extensions/filters/http/cache/cache_headers_utils.h

```diff
@@ -128,16 +124,24 @@ class VaryHeader {
   // Parses the header names that are in the Vary header value.
   static std::vector<std::string> parseHeaderValue(const Http::HeaderEntry* vary_header);
 
+  // Parses the allowlist from the Cache Config into the object's private allowlist_.
+  VaryHeader(const Protobuf::RepeatedPtrField<envoy::type::matcher::v3::StringMatcher>& allowlist);
+
+  // Checks if the headers contain an allowed value in the Vary header.
+  bool isAllowed(const Http::ResponseHeaderMap& headers) const;
+
   // Returns a header map containing the subset of the original headers that can be varied from the
   // request.
-  static Http::RequestHeaderMapPtr
-  possibleVariedHeaders(const std::vector<Matchers::StringMatcherPtr>& allowlist,
-                        const Http::RequestHeaderMap& request_headers);
-
+  Http::RequestHeaderMapPtr
+  possibleVariedHeaders(const Http::RequestHeaderMap& request_headers) const;
   // Parses the allowlist of matching rules that define header names that can be used to create
   // varied responses.
   static std::vector<Matchers::StringMatcherPtr> parseAllowlist(
```

Comment:
ditto -- can this static be made private? or just inlined into the constructor?

Reply:
Yes, this one can be private (or restructured) as it's only used by the constructor.

Reply:
I moved it inside the constructor.